server/tinylicious/src/runner.ts

```diff
@@ -82,18 +84,24 @@ export class TinyliciousRunner implements utils.IRunner {
     /**
      * Ensure provided port is free
      */
-    private async ensurePortIsFree(): Promise<void> {
+    private async ensurePortIsFree(exitOnPortConflict: boolean): Promise<boolean> {
         // If port is a named pipe resolve immediately
         if (typeof this.port === "string") {
-            return;
+            return true;
         }
 
         const freePort = await detect(this.port);
         if (this.port === freePort) {
-            return;
+            return true;
         }
 
-        throw new Error(`Port: ${this.port} is occupied. Try port: ${freePort}`);
+        const errorMessage = `Port: ${this.port} is occupied. Try port: ${freePort}`;
+
+        if (exitOnPortConflict) {
```

Comment:
Should it always unconditionally throw, and start() just catch and print error message and return?